### PR TITLE
VTCCA-3042: added clientsPropertyLink projection

### PR DESCRIPTION
**Pull request by habeeb01**

### CODE REVIEW

File: app/controllers/PropertyLinkingController.scala [hunk 1]
```diff
@@ -105,10 +106,22 @@ class PropertyLinkingController @Inject()(
...
+        propertyLinkService
+          .getClientsPropertyLink(submissionId)
+          .fold(NotFound("client property link not found")) { pl => Ok(Json.toJson(PropertiesView(pl.authorisation, Nil))) }
+      }
+      case "clientsPropertyLink" => {
+        propertyLinkService
+          .getClientsPropertyLink(submissionId)
+          .fold(NotFound("client property link not found")) { pl => Ok(Json.toJson(pl.authorisation)) }
+      }
+      case _ => throw new IllegalArgumentException(s"unsupported projection: $projection")
```
Comment (PeterPerhac) [inline app/controllers/PropertyLinkingController.scala]: wouldn't this produce a 500 response? I think the projection should be an enum of possible / supported values and fail with 400 if the received value is not supported.

Comment (habeeb01) [reply app/controllers/PropertyLinkingController.scala]: I have updated it and I am now returning a 501 instead. It is how we have handled this in other places.

**Comment by PeterPerhac** [comment]
👍

**Comment by darrell-plant** [comment]
👍

**Pull request merged**